File: app/src/main/java/com/ericho/coupleshare/adapter/TestMainRecyclerAdapter.java
package com.ericho.coupleshare.adapter;

import android.content.Context;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;


import com.ericho.coupleshare.R;
import com.ericho.coupleshare.setting.model.AppTestBo;

import java.util.List;

import butterknife.BindView;
import butterknife.ButterKnife;
import io.reactivex.Observable;
import io.reactivex.subjects.PublishSubject;

/**
 * Created by EricH on 19/4/2017.
 */

public class TestMainRecyclerAdapter extends BaseRecyclerAdapter<AppTestBo, TestMainRecyclerAdapter.ViewHolder> {
    private final PublishSubject<AppTestBo> onClickSubject = PublishSubject.create();

    public TestMainRecyclerAdapter(Context context, List<AppTestBo> items) {
        super(context, items);
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(getContext()).inflate(R.layout.grid_change_server, parent, false);
        ViewHolder viewHolder = new ViewHolder(view);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(ViewHolder holder, int position) {
        AppTestBo o = getItems().get(position);
        holder.imageView.setOnClickListener(v -> onClickSubject.onNext(o));
        holder.textView.setOnClickListener(v -> onClickSubject.onNext(o));
        holder.textView.setText(o.getDisplayName());

    }


    @Override
    public int getItemCount() {
        return items.size();
    }

    public Observable<AppTestBo> getPositionClicks() {
        return onClickSubject;
    }

    static class ViewHolder extends RecyclerView.ViewHolder {
        @BindView(R.id.text1)
        TextView textView;
        @BindView(R.id.image)
        ImageView imageView;

        public ViewHolder(View view) {
            super(view);
            ButterKnife.bind(this, view);
        }
    }
}

File: app/src/main/java/com/ericho/coupleshare/adapter/TestMainRecyclerAdapter.kt
package com.ericho.coupleshare.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.AdapterView
import android.widget.ImageView
import android.widget.TextView
import com.ericho.coupleshare.R
import com.ericho.coupleshare.setting.model.AppTestBo

/**
 * Created by steve_000 on 8/7/2017.
 * for project CoupleShare
 * package name com.ericho.coupleshare.adapter
 */
class TestMainRecyclerAdapter constructor(context: Context,items:List<AppTestBo>):BaseRecyclerAdapter<AppTestBo, TestMainRecyclerAdapter.ViewHolder>(context, items) {



    var onItemClickListener: AdapterView.OnItemClickListener? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val view = LayoutInflater.from(getContext()).inflate(R.layout.grid_change_server, parent, false)
        val viewHolder = ViewHolder(view)
        return viewHolder
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val o = items!![position]
        holder.imageView.setOnClickListener { _ -> onItemClickListener?.onItemClick(null,holder.imageView,position,getItemId(position)) }
        holder.textView.setOnClickListener { _ -> onItemClickListener?.onItemClick(null,holder.imageView,position,getItemId(position)) }
        holder.textView.text = o.displayName

    }



    class ViewHolder(view: View) : RecyclerView.ViewHolder(view) {
        var textView = view.findViewById(R.id.text1) as TextView
        var imageView = view.findViewById(R.id.image) as ImageView

    }


}